### DISCUSSION TITLE
ver 6.0.0 proper way of using actionRow Snippet

### DISCUSSION THREAD

**Discussion by sysmat** [mskocik/svelty-picker#181] [Q&A] [answered]
What is proper way of using actionRow, maybe should be in documentation

**Comment by mskocik** [comment]
![image](https://github.com/user-attachments/assets/ad868d5a-70fb-40e1-b851-07c555f6a06d)

This is the default `actionRow` rendered. If you want to customize it, you can by providing your own snippet. There is default implementation code displayed in documentation. Does it answer your question?

**Reply by sysmat** [reply]
I'm struggeling with typescript

- before

```html
<SveltyPicker startView={3} mode="time" format="G:i" formatType="php" autocommit={false} pickerOnly onChange={(value) => {childrenEditor!.addChild(value as string)}}>
  <svelte:fragment slot="action-row"   let:onConfirm  >
         <div class="flex justify-end mt-2">
               <Button
                        variant="outline"
                        onclick={onConfirm}>
                                  {$t('editor.voting.structureEditor.rootElement.timeButtonConfirmText')}
                </Button>
           </div>
     </svelte:fragment>
</SveltyPicker>
```

- after, how to bind  specific prop in `actionRow?: Snippet<[autocloseSupported: boolean, todayBtnClasses: string, clearBtnClasses: string, onCancel: Function, onConfirm: Function, onClear: Function, onToday: Function, isTodayDisabled: boolean | null, i18n: i18nType, currentMode: string]>;`

```html
<SveltyPicker startView={3} mode="time" format="G:i" formatType="php" autocommit={false} pickerOnly onChange={(value) => {childrenEditor!.addChild(value as string)}}>
      {#snippet actionRow(onConfirm)}
            <div class="flex justify-end mt-2 bg-red-500">
                       <Button
                                variant="outline"
                                 onclick={onConfirm}>
                              {$t('editor.voting.structureEditor.rootElement.timeButtonConfirmText')}
                        </Button>
            </div>
       {/snippet}                                                
</SveltyPicker>
```

**Reply by mskocik** [reply]
You just use the whole snippet definition, even though you don't need the rest of props. They are specified, so that you are able to recreate default behavior, if needed.

```svelte

{#snippet myCustomActionRow(autocloseSupported, todayBtnClasses, clearBtnClasses, onCancel, onConfirm, onClear, onToday, isTodayDisabled, i18n, currentMode)}
    <div class="flex justify-end mt-2 bg-red-500">
      <Button
        variant="outline"
        onclick={onConfirm}
      >
        {$t('editor.voting.structureEditor.rootElement.timeButtonConfirmText')}
      </Button>
    </div>
{/snippet}  

<SveltyPicker
  startView={3}
  mode="time"
  format="G:i"
  formatType="php"
  autocommit={false}
  pickerOnly
  onChange={(value) => {childrenEditor!.addChild(value as string)}}
  actionRow={myCustomActionRow}
></SveltyPicker>
```

**Reply by sysmat** [reply]
@mskocik thx

**Reply by mskocik** [reply]
I think you marked wrong answer :-)

**Reply by sysmat** [reply]
I'm new to svelte 5, but if lib define snippet, shouldn't user than use `@render`, or I'm missing something

**Reply by sysmat** [reply]
> There is 1 available snippet actionRow to override

aha, then user of this lib define `#snippet action_row`

**Reply by mskocik** [reply, marked as answer]
I've updated the code I wrote above 👆. I didn't realize that snippet definition was incorrect.

**Reply by sysmat** [reply]
yes, my college notice me

**Reply by sysmat** [reply]
could be `[autocloseSupported: boolean, todayBtnClasses: string, clearBtnClasses: string, onCancel: Function, onConfirm: Function, onClear: Function, onToday: Function, isTodayDisabled: boolean | null, i18n: import("../i18n/index.js").i18nType, currentMode: string]`

this as new type with ts Partial?

**Reply by mskocik** [reply]
Yes, I was thinking about it, that it could be more practical, especially when someone is not interested in all props.

**Reply by mskocik** [reply]
> could be `[autocloseSupported: boolean, todayBtnClasses: string, clearBtnClasses: string, onCancel: Function, onConfirm: Function, onClear: Function, onToday: Function, isTodayDisabled: boolean | null, i18n: import("../i18n/index.js").i18nType, currentMode: string]`
> 
> this as new type with ts Partial?

I just released new version, where this is implemented

**Reply by sysmat** [reply]
thx, I'll look it

**Reply by sysmat** [reply]
aha, you don't use typescript, you are using jsdoc.

```typescript
type ActionRowType = {
		autocloseSupported: boolean;
		todayBtnClasses: string;
		clearBtnClasses: string;
		onCancel: Function;
		onConfirm: Function;
		onClear: Function;
		onToday: Function;
		isTodayDisabled: boolean | null;
		i18n: string;
		currentMode: string;
	};

type ActionRowPartialType = Partial<ActionRowType >;
```
